src/ophyd_async/tango/core/_tango_transport.py

```diff
@@ -735,7 +735,10 @@ def set_callback(self, callback: Callback | None) -> None:
 
         if callback:
             try:
-                assert not self.proxies[self.read_trl].has_subscription()  # type: ignore
+                # TODO: remove assert, I fail to get the test to work
+                # as soon as move the assert into a separate RunTime error
+                # the `test_tango_sim` fails
+                assert not self.proxies[self.read_trl].has_subscription()  # type: ignore  # noqa: S101
```

Comment:
@burkeds any idea why this is the case?

Reply:
Hmm, I gave it a another go, using the same logic as I use for `aioca` and `p4p`, and that worked. When I moved the AssertionError, that is re-raised as RuntimeError into a separate if-statement, a bunch of tests failed.
fixed in 8592cbd7aa8780e048922bad95ab9776ab19eac0